### PR TITLE
fix: Resend exception when errors occured in SubmitAsync

### PR DESCRIPTION
**Pull request by ddubuc**
Main goal is to add more logs and a better management of communication error. Only Mock could reproduce that expected error. In relation with Extensions.Csharp https://github.com/aneoconsulting/ArmoniK/issues/678

### CODE REVIEW

**Comment by ddubuc** [comment]
link to issue https://github.com/aneoconsulting/ArmoniK/issues/977

File: Client/src/Unified/Services/Submitter/Service.cs [hunk 1]
```diff
@@ -139,34 +141,91 @@ public class Service : AbstractClientService, ISubmitterService
...
+                                        SessionService.SubmitTasksWithDependencies(blockRequestList.Select(x => new
+                                                                                                             Tuple<string, byte[], IList<string>>(x.ResultId.ToString(),
+                                                                                                                                                  x.Payload!.Serialize(),
+                                                                                                                                                  null)),
+                                                                                   1);
+
+                                      var ids            = taskIds.ToList();
+                                      var mapTaskResults = SessionService.GetResultIds(ids);
+                                      var taskIdsResultIds = mapTaskResults.ToDictionary(result => result.ResultIds.Single(),
+                                                                                         result => result.TaskId);
```
Comment (dbrasseur-aneo) [inline Client/src/Unified/Services/Submitter/Service.cs:170]: Are we sure each task has one, and only one, result ?

File: Client/src/Unified/Services/Submitter/Service.cs [hunk 2]
```diff
@@ -139,34 +141,91 @@ public class Service : AbstractClientService, ISubmitterService
...
+
+                                        requestTaskMap_.PutResponse(blockRequest.SubmitId,
+                                                                    pairTaskIdResultId.Value);
+                                      }
+
+                                      if (ids.Count() > taskIdsResultIds.Count)
+                                      {
+                                        Logger?.LogWarning("Fail to submit all tasks in once, retry with missing tasks");
+
+                                        throw new Exception("Fail to submit all tasks in once. Retrying...");
```
Comment (dbrasseur-aneo) [inline Client/src/Unified/Services/Submitter/Service.cs:191]:
```suggestion
                                        Logger?.LogWarning("Fail to submit all tasks at once, retry with missing tasks");

                                        throw new Exception("Fail to submit all tasks at once. Retrying...");
```

**Review by dbrasseur-aneo — CHANGES_REQUESTED** [review]

File: Client/src/Unified/Services/Submitter/Service.cs (hunk 1 above)
Comment (aneojgurhem) [reply Client/src/Unified/Services/Submitter/Service.cs:170]: By construction, yes

**Review by aneojgurhem — COMMENTED** [review]
What issue do this contribution fix ? Can we reproduce it with a test case ?

File: Client/src/Unified/Services/Submitter/Service.cs (hunk 1 above)
Comment (ddubuc) [reply Client/src/Unified/Services/Submitter/Service.cs:170]: yep only one result for one task. Multi results aren't yet implemented

**Comment by ddubuc** [comment]
@aneojgurhem : Main goal is to add more logs and a better management of communication error. Only Mock could reproduce that expected error. In relation with Extensions.Csharp https://github.com/aneoconsulting/ArmoniK/issues/541

**Comment by aneojgurhem** [comment]
> @aneojgurhem : Main goal is to add more logs and a better management of communication error. Only Mock could reproduce that expected error. In relation with Extensions.Csharp [aneoconsulting/ArmoniK#678](https://github.com/aneoconsulting/ArmoniK/issues/678)

So that should be mentioned in the PR description !

File: Client/src/Unified/Services/Submitter/Service.cs [hunk 3]
```diff
@@ -56,6 +56,8 @@ namespace ArmoniK.DevelopmentKit.Client.Unified.Services.Submitter;
 [MarkDownDoc]
 public class Service : AbstractClientService, ISubmitterService
 {
+  private const int MaxRetries = 5;
```
Comment (ngruelaneo) [inline Client/src/Unified/Services/Submitter/Service.cs:59]: this value should be a parameter to be able to modify it if needed by the extensions.csharp user

**Review by ngruelaneo — COMMENTED** [review]
Will it be possible to have some tests ?

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**